### PR TITLE
KNOX-3020 - Introducing the 'type' metadata for Knox Tokens

### PR DESCRIPTION
**Pull request by smolnar82**
## What changes were proposed in this pull request?

Introduced new metadata called `type`, this can be any of:
- JWT (default)
- KNOXSSO_COOKIE
- CLIENT_ID

The default is `JWT`. The `KNOXSSO_COOKIE` is set when tokens are participating in KNOX SSO Cookie management we introduced in #797 .
`CLIENT_ID` can be set via a KnoxToken API call with metadata name, `type`. For instance:
```
curl -iku admin:admin-password https://localhost:8443/gateway/sandbox/knoxtoken/api/v2/token?md_type=CLIENT_ID
```
The Token Management UI is updated: instead of adding new columns per type, only a one column exists and the relevant type icon is displayed.

## How was this patch tested?
Ran build/unit tests and deployed Knox. Then I 
- generated a regular JWT token on the Token Generation UI
- enabled the KnoxSSO cookie management feature and re-logged in to Knox Home
- used the above `curl` request to generate a `CLIENT_ID` type token

Confirmed the tokens were created and they were displayed properly on the Token Management UI:

Screenshot 2024-03-13 at 19 03 50

### CODE REVIEW

File: gateway-spi/src/main/java/org/apache/knox/gateway/services/security/token/TokenMetadata.java [hunk 1]
```diff
@@ -142,6 +139,32 @@ public Instant getLastUsedAt() {
     return lastUsedAt == null ? null : Instant.parse(lastUsedAt);
   }
 
+  public void setType(String type) {
```
Comment (moresandeep) [inline gateway-spi/src/main/java/org/apache/knox/gateway/services/security/token/TokenMetadata.java:142]: Can we use just enum Type for simplicity?

**Review by moresandeep — APPROVED** [review]
Few minor things but overall 👍 love the icons :)

File: gateway-spi/src/main/java/org/apache/knox/gateway/services/security/token/TokenMetadata.java (hunk 1 above)
Comment (smolnar82) [reply gateway-spi/src/main/java/org/apache/knox/gateway/services/security/token/TokenMetadata.java:142]: Ack.

File: gateway-spi/src/main/java/org/apache/knox/gateway/services/security/token/TokenMetadata.java [hunk 2]
```diff
@@ -32,14 +32,18 @@
...
+
   public static final String USER_NAME = "userName";
   public static final String COMMENT = "comment";
   public static final String ENABLED = "enabled";
   public static final String PASSCODE = "passcode";
   public static final String CREATED_BY = "createdBy";
-  public static final String KNOX_SSO_COOKIE = "knoxSSOCookie";
   public static final String LAST_USED_AT = "lastUsedAt";
-  private static final List<String> KNOWN_MD_NAMES = Arrays.asList(USER_NAME, COMMENT, ENABLED, PASSCODE, CREATED_BY, KNOX_SSO_COOKIE, LAST_USED_AT);
+  public static final String TYPE = "type";
```
[the remark this answers is not in the document]
Comment (smolnar82) [reply gateway-spi/src/main/java/org/apache/knox/gateway/services/security/token/TokenMetadata.java:45]: The enum is TokenMetadata.Type. Let me figure out something.

[the remark this answers is not in the document]
Comment (smolnar82) [reply gateway-spi/src/main/java/org/apache/knox/gateway/services/security/token/TokenMetadata.java:41]: Done.

**Review by moresandeep — APPROVED** [review]
looks good, thanks!

**Pull request merged**